Core/include/Acts/Surfaces/CurvilinearSurface.hpp

```diff
@@ -0,0 +1,185 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Definitions/TrackParametrization.hpp"
+#include "Acts/Surfaces/PlaneSurface.hpp"
+#include "Acts/Utilities/JacobianHelpers.hpp"
+#include "Acts/Utilities/VectorHelpers.hpp"
+
+#include <iomanip>
+#include <ios>
+#include <ostream>
+#include <string>
+
+namespace Acts {
+
+/// @brief Utility class for curvilinear surfaces
+///
+class CurvilinearSurface {
```

Comment:
This could potentially use the surface concept that I added a while back

Reply:
nice! I took a look and I fear it requires more functions than necessary here